platform/windows/display_server_windows.cpp

```diff
@@ -2158,7 +2158,10 @@ void DisplayServerWindows::popup_close(WindowID p_window) {
 		WindowID win_id = E->get();
 		popup_list.erase(E);
 
-		_send_window_event(windows[win_id], DisplayServerWindows::WINDOW_EVENT_CLOSE_REQUEST);
+		if (win_id != p_window) {
+			// Only request close on related windows, not this window.  We are already processing it.
+			_send_window_event(windows[win_id], DisplayServerWindows::WINDOW_EVENT_CLOSE_REQUEST);
+		}
```

Comment:
This is likely relevant for X11 and macOS as well (need testing, code is almost same).

Reply:
I just put that in the notes above, since I am not set up to test Linux (yet) unfortunately.  Is there something else I could have done to make sure this PR gets to the right "other platforms" people?   Obviously you caught it this time, just wondering about future.